app/src/androidTest/java/com/github/orkest/View/auth/SignInTest.kt

```diff
@@ -88,33 +94,80 @@ class SignInTest {
 
+        // Mock credentials
+        val username = "test"
+        val email = "test@test.ch"
 
+        // Save and load user credentials
+        saveUserCredentials(context, username, email)
+        val (savedUsername, savedEmail) = loadUserCredentials(context)
 
+        // Check if the provided user credentials do not match the saved ones
+        assertEquals(username, savedUsername)
+        assertEquals(email, savedEmail)
 
+        // Clear SharedPreferences after each test
+        val sharedPref = context.getSharedPreferences("user_credentials", Context.MODE_PRIVATE)
+        sharedPref.edit().clear().apply()
+    }
 
-    /////////////
-    /*
+    /**
+     * Checks if credentials are present in SharedPreferences
+     */
     @Test
-    fun existingUsernameLaunchesMain(){
-        Intents.init()
-        composeTestRule.onNodeWithText("Current Username")
-            .performTextInput(MockAuthViewModel.EXISTING_USER)
-
-        composeTestRule.onNodeWithText("Sign In").performClick()
-        Intents.intended((IntentMatchers.hasComponent(MainActivity::class.java.name)))
-        Intents.release()
+    fun correctlyChecksOffline() {
+        val context: Context = ApplicationProvider.getApplicationContext()
+
+        // Mock credentials
+        val username = "test"
+        val email = "test@test.ch"
+
+        // Save user credentials
+        saveUserCredentials(context, username, email)
+
+        //check offline
+        assertEquals(true, isSignedInOffline(context))
+
+        // Clear SharedPreferences after each test
+        val sharedPref = context.getSharedPreferences("user_credentials", Context.MODE_PRIVATE)
+        sharedPref.edit().clear().apply()
     }
 
+    /**
+     * Correctly cleans the elements in SharedPreferences
+     */
     @Test
-    fun noPermissionDisplaysError(){
-        composeTestRule.onNodeWithText("Current Username")
-            .performTextInput(MockAuthViewModel.NO_PERMISSIONS)
+    fun correctlyCleansCache(){
+        val context: Context = ApplicationProvider.getApplicationContext()
+
+        // Mock credentials
+        val username = "test"
+        val email = "test@test.ch"
 
-        composeTestRule.onNodeWithText("Sign In").performClick()
+        // Save user credentials
+        saveUserCredentials(context, username, email)
 
-        composeTestRule.onNodeWithText("No permissions for this user!").assertIsDisplayed()
-    }*/
+        // Clear user credentials
+        cleanSigningCache(context)
+
+        // Load the new values
+        val (savedUsername, savedEmail) = loadUserCredentials(context)
+
+        // Check if the provided user credentials are empty
+        assertEquals(null, savedUsername)
+        assertEquals(null, savedEmail)
+
+        // Clear SharedPreferences after each test
+        val sharedPref = context.getSharedPreferences("user_credentials", Context.MODE_PRIVATE)
+        sharedPref.edit().clear().apply()
```

Comment:
Nice commenting each part of the test, makes it much easier to understand

Reply:
Thanks